pkg/comp-functions/functions/vshnkeycloak/deploy.go

```diff
@@ -0,0 +1,373 @@
+			params.Backup.DeletionProtection = false
+		}
+	}
+
+	pg := &vshnv1.XVSHNPostgreSQL{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: comp.GetName() + pgInstanceNameSuffix,
+		},
+		Spec: vshnv1.XVSHNPostgreSQLSpec{
+			Parameters: *params,
+			ResourceSpec: xpv1.ResourceSpec{
+				WriteConnectionSecretToReference: &xpv1.SecretReference{
+					Name:      pgSecretName,
+					Namespace: comp.GetInstanceNamespace(),
+				},
+			},
+		},
+	}
+
+	err := common.CreateNetworkPolicy([]string{comp.GetInstanceNamespace()}, pg.GetInstanceNamespace(), pg.GetName()+"-keycloak", svc)
+	if err != nil {
+		return err
+	}
+
+	return svc.SetDesiredComposedResource(pg)
+}
+
+func addRelease(ctx context.Context, svc *runtime.ServiceRuntime, comp *vshnv1.VSHNKeycloak, adminSecret string) error {
+	release, err := newRelease(ctx, svc, comp, adminSecret)
+	if err != nil {
+		return err
+	}
+
+	return svc.SetDesiredComposedResource(release)
+}
+
+func getResources(ctx context.Context, svc *runtime.ServiceRuntime, comp *vshnv1.VSHNKeycloak) (common.Resources, error) {
+	plan := comp.Spec.Parameters.Size.GetPlan(svc.Config.Data["defaultPlan"])
+
+	resources, err := utils.FetchPlansFromConfig(ctx, svc, plan)
+	if err != nil {
+		err = fmt.Errorf("cannot fetch plans from the composition config, maybe they are not set: %w", err)
+		return common.Resources{}, err
+	}
+
+	res := common.GetResources(&comp.Spec.Parameters.Size, resources)
+
+	return res, nil
+}
+
+func newValues(ctx context.Context, svc *runtime.ServiceRuntime, comp *vshnv1.VSHNKeycloak, adminSecret string) (map[string]any, error) {
+	values := map[string]any{}
+
+	cd, err := svc.GetObservedComposedResourceConnectionDetails(comp.GetName() + pgInstanceNameSuffix)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := getResources(ctx, svc, comp)
+	if err != nil {
+		return nil, err
+	}
+
+	plan := comp.Spec.Parameters.Size.GetPlan(svc.Config.Data["defaultPlan"])
+	nodeSelector, err := utils.FetchNodeSelectorFromConfig(ctx, svc, plan, comp.Spec.Parameters.Scheduling.NodeSelector)
+	if err != nil {
+		return values, fmt.Errorf("cannot fetch nodeSelector from the composition config: %w", err)
+	}
+
+	fqdn := ""
+	strictHostname := false
+	if comp.Spec.Parameters.Service.FQDN != "" {
+		fqdn = comp.Spec.Parameters.Service.FQDN
+		strictHostname = true
+	}
+
+	extraEnv, err := toYaml([]map[string]any{
+		{
+			"name": "JAVA_OPTS",
+			"value": `-XX:+UseContainerSupport
+-XX:MaxRAMPercentage=50.0
+-Djava.net.preferIPv4Stack=true
+-Djava.awt.headless=true
+-Djgroups.dns.query={{ include "keycloak.fullname" . }}-headless`,
+		},
+		{
+			"name": "KEYCLOAK_ADMIN_PASSWORD",
+			"valueFrom": map[string]any{
+				"secretKeyRef": map[string]any{
+					"name": adminSecret,
+					"key":  "password",
+				},
+			},
+		},
+		{
+			"name":  "KEYCLOAK_ADMIN",
+			"value": "admin",
+		},
+		{
+			"name":  "KC_HOSTNAME",
+			"value": fqdn,
+		},
+		{
+			"name":  "KC_HOSTNAME_STRICT",
```

Comment:
This is for HTTP only. HTTPS is strict by default.

Reply:
I currently keep it, because if no name is specified then keycloak refuses to start entirely.